### DISCUSSION TITLE
[2023-11-20] Team GUI Demo Day

### DISCUSSION THREAD

**Discussion by farmaazon** [enso-org/enso#8343] [Weekly Demo Day]
* Implementing using alias analysis and almost finished it: #8316 

![Screenshot from 2023-11-16 15-50-27](https://github.com/enso-org/enso/assets/3919101/691aac6c-1951-4988-a338-4449c0413ef7)

* Next week plans: implement #7787 but first fix all discovered issues.

**Comment by MichaelMauderer** [comment]
* Finished #8298
* Finished #8303
* Implemented #8341

[Peek 2023-11-20 13-50.webm](https://github.com/enso-org/enso/assets/1428930/550ea722-a8fa-46b1-820e-43e00e1c396b)

**Comment by somebody1234** [comment]
- spent most of last week on website2 (bugfixes, minor touchups)
- i believe wojciech was mostly working on changes to text, colors, the final section, and the animations for small screens

PRs merged:
- #8310
- #8311
- #8326
- #8327

a reminder that GUI2 should be basically usable for testing with book clubs! especially with the PRs merged over the past week

**Comment by vitvakatu** [comment]
Finishing imports: https://github.com/enso-org/enso/pull/8315.

Because of the number of corner cases and occasional regressions on develop, it took longer than estimated.

https://github.com/enso-org/enso/assets/6566674/ecd3d032-286b-4bfa-8dc6-96e178f595f3

Now starting to work on https://github.com/enso-org/enso/issues/8260

**Comment by Frizi** [comment]
- Implemented displaying argument widgets, both for existing and missing ones. #8339 
Connecting to placeholders is not fully implemented yet, and it is a separate task.

https://github.com/enso-org/enso/assets/919491/5d99ab3f-07db-48d5-9ba7-556a229226f7

Nested argument lists are also handled nicely.
<img width="396" alt="image" src="https://github.com/enso-org/enso/assets/919491/c6fd6248-a685-4650-b4bf-63ee768640aa">

**Comment by mwu-tow** [comment]
* Fixed macOS builds & releases (https://github.com/enso-org/enso/pull/8284);

Ongoing stuff:
* post-graal-bump backend release issue;
* recent 7.4.0 release broke our build-script wrapper;
* fixing NSIS installer on newer electron-builder.

**Comment by kazcw** [comment]
Finished new AST (#8263) implementation. Will PR porting app logic to new types today. Using the new AST to edit nodes will be in a follow-up PR.

The new AST allows the application to start its analysis from a higher level of abstraction:
<img width="1264" alt="image" src="https://github.com/enso-org/enso/assets/1047859/d870ae63-bdfa-419b-b1cb-2a40d66601f7">

**Comment by JaroslavTulach** [comment]
Major work be engine-ers: 

- Improving Component Browser - #8148 
-  **16% speedup** of Hello World - #8207
- Upgrade to latest GraalVM - #7991
